### PR TITLE
Stripe migration

### PR DESCRIPTION
**Pull request by dem4ron**
Screenshot 2023-07-11 at 14 54 36
Screenshot 2023-07-11 at 14 55 08

### Lazy load cases:
Screenshot 2023-08-07 at 16 11 26
Screenshot 2023-08-07 at 16 12 36

### CODE REVIEW

File: app/commands/payments/stripe/payment_intent/create_for_payment.rb [hunk 1]
```diff
@@ -8,7 +8,9 @@ def call
       customer: customer_id,
       amount: amount_in_cents,
       currency: 'usd',
-      setup_future_usage: 'off_session'
```
Comment (iHiD) [inline app/commands/payments/stripe/payment_intent/create_for_payment.rb:11]: Is this deprecated?

File: app/javascript/components/donations/ExercismStripeElements.tsx [hunk 2]
```diff
@@ -20,12 +20,9 @@ function load() {
...
 const options = {
-  fonts: [
-    {
-      cssSrc:
-        'https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&display=swap',
-    },
-  ],
+  mode: 'payment',
+  amount: 1234,
+  currency: 'usd',
```
Comment (iHiD) [inline app/javascript/components/donations/ExercismStripeElements.tsx]: This seems odd...

File: app/javascript/components/donations/StripeForm.tsx [hunk 3]
```diff
@@ -156,19 +163,20 @@ export function StripeForm({
...
-        }
-      )
+      // const cardElement = elements.getElement(CardElement)!
 
-      if (payload.error) {
+      const { error } = await stripe.confirmPayment({
+        elements,
+        clientSecret: paymentIntent.clientSecret,
+        confirmParams: {
+          return_url: 'http://local.exercism.io:3020/donate',
```
Comment (iHiD) [inline app/javascript/components/donations/StripeForm.tsx]: This isn't what you want.

File: app/javascript/components/donations/StripeForm.tsx [hunk 4]
```diff
@@ -246,22 +254,13 @@ export function StripeForm({
...
-            disabled={
-              !notARobot ||
-              processing ||
-              !cardValid ||
-              succeeded ||
-              (!userSignedIn && email.length === 0)
-            }
-          >
+          /> */}
+          <button className="btn-primary btn-s" type="submit">
```
Comment (iHiD) [inline app/javascript/components/donations/StripeForm.tsx]: Why have we lost the `disabled` stuff?

File: app/commands/payments/stripe/payment_intent/create_for_payment.rb (hunk 1 above)
Comment (dem4ron) [reply app/commands/payments/stripe/payment_intent/create_for_payment.rb:11]: it must be `nil` for this to work.

Comment (iHiD) [reply app/commands/payments/stripe/payment_intent/create_for_payment.rb:11]: If this isn't documented (and I can't find it), could you open a support request with Stripe to double-check this please.

File: app/javascript/components/donations/footer-form/stripe-form-modal/DonationForm.tsx
Comment (dem4ron) [inline app/javascript/components/donations/footer-form/stripe-form-modal/DonationForm.tsx:1]: Let's just delete this

File: app/javascript/components/insiders/InsidersStatus.tsx [hunk 5]
```diff
@@ -137,8 +137,9 @@ export default function Status({
       >
         <ModalHeader period={'lifetime'} />
         <hr className="mb-32 border-borderColor5 -mx-48" />
-        <ExercismStripeElements>
+        <ExercismStripeElements amount={499}>
```
Comment (ErikSchierboom) [inline app/javascript/components/insiders/InsidersStatus.tsx]: Could we pass this amount in from the backend?

File: app/javascript/components/donations/stripe-form/utils.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      return 'Subscribe to Premium'
+    default:
+      return ''
+  }
+}
+
+export function generateIntervalText(paymentIntent: PaymentIntentType): string {
+  switch (paymentIntent) {
+    case 'premium_monthly_subscription':
+      return `month`
```
Comment (ErikSchierboom) [inline app/javascript/components/donations/stripe-form/utils.ts]:
```suggestion
      return 'month'
```

File: app/javascript/components/donations/stripe-form/utils.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    case 'premium_yearly_subscription':
+      return `year`
```
Comment (ErikSchierboom) [inline app/javascript/components/donations/stripe-form/utils.ts]:
```suggestion
      return 'year'
```

**Comment by iHiD** [comment]
I've merged this into #5862 now. So I'll close it here. Nice work :)

**Pull request closed without merging**